File: src/main/java/com/HOIIVUtils/clauzewitz/data/country/Character.java
package com.HOIIVUtils.clauzewitz.data.country;

import com.HOIIVUtils.clauzewitz.localization.Localizable;
import org.jetbrains.annotations.NotNull;

import java.util.Collection;
import java.util.List;
import java.util.Map;

public class Character implements Localizable {
    String id;

    @Override
    public @NotNull Map<Property, String> getLocalizableProperties() {
        return Map.of(Property.NAME, id);
    }

    @Override
    public @NotNull Collection<? extends Localizable> getLocalizableGroup() {
        return List.of(this); // todo improve
    }
}

File: src/main/scala/com/hoi4utils/hoi4/history/countries/Character.scala
package com.hoi4utils.hoi4.history.countries

import com.hoi4utils.hoi4.localization.{Localizable, Property}

import scala.collection.mutable

class Character extends Localizable {
  private val id: String = null

  override def localizableProperties: Map[Property, String] = Map(Property.NAME -> id)

  override def getLocalizableGroup: Iterable[? <: Localizable] = List(this) // todo improve
}

